packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -0,0 +1,559 @@
+///               );
+///             },
+///           ),
+///         ),
+///       ),
+///     );
+///   }
+/// }
+/// ```
+/// {@end-tool}
+class AutocompleteCore<T> extends StatefulWidget {
+  /// Create an instance of AutocompleteCore.
+  ///
+  /// [autocompleteController], [buildField], and [buildResults] must not be
+  /// null.
+  const AutocompleteCore({
+    this.autocompleteController,
+    @required this.buildField,
+    @required this.buildResults,
+    this.options,
+    this.onSelected,
+  }) : assert(
+         autocompleteController != null || options != null,
+         'If not providing autocompleteController, options must be passed.',
+       ),
+       assert(
+         autocompleteController == null || options == null,
+         'No need to pass options if providing an AutocompleteController',
+       ),
+       assert(buildField != null),
+       assert(buildResults != null);
+
+  /// The controller that provides access to the main autocomplete state and
+  /// logic.
+  ///
+  /// The owner of autocompleteController must call
+  /// [AutocompleteController.dispose] on this when it's no longer needed.
+  final AutocompleteController<T> autocompleteController;
+
+  /// Builds the field that is used to input the query.
+  final AutocompleteFieldBuilder buildField;
+
+  /// Builds the selectable results of filtering.
+  final AutocompleteResultsBuilder<T> buildResults;
+
+  /// Called when a result is selected by the user.
+  ///
+  /// This method is used to update the query field to reflect the selection, so
+  /// if implemented, it should set `textEditingController.value`.
+  final OnSelectedAutocomplete<T> onSelected;
+
+  /// All possible options that can be selected.
+  ///
+  /// If passing an AutocompleteController, use
+  /// [AutocompleteController.options] instead.
+  final List<T> options;
+
+  @override
+  _AutocompleteCoreState<T> createState() =>
+      _AutocompleteCoreState<T>();
+}
+
+class _AutocompleteCoreState<T> extends State<AutocompleteCore<T>> {
+  final GlobalKey _fieldKey = GlobalKey();
+  AutocompleteController<T> _autocompleteController;
+  T _selection;
+
+  // The OverlayEntry containing the results.
+  OverlayEntry _cachedFloatingResults;
+  OverlayEntry get _floatingResults {
+    if (_cachedFloatingResults != null) {
+      return _cachedFloatingResults;
+    }
+
+    assert(_fieldKey.currentContext != null);
+    final RenderBox renderBox = _fieldKey.currentContext.findRenderObject() as RenderBox;
+    final Offset offset = renderBox.localToGlobal(Offset.zero);
+    _cachedFloatingResults = OverlayEntry(
+      builder: (BuildContext context) {
+        return Positioned(
+          top: offset.dy + renderBox.size.height,
+          left: offset.dx,
+          width: renderBox.size.width,
+          child: widget.buildResults(
+            context,
+            _onSelected,
+            _autocompleteController.results.value,
+          ),
+        );
+      },
+    );
```

Comment:
With this arrangement for buildResults, the size of the OverlayEntry is specified by the user, but the position is not. 

If the user directly passes something that needs height constraints, like a ListView, without wrapping it in a SizedBox with a height, it will fail.

The OverlayEntry is always full width. It's also always placed directly below the query field. This isn't changeable by the user (without building something custom with AutocompleteController).

Reply:
Moving this comment to a valid location after code changes.